Modules/Sources/Yosemite/PointOfSale/Eligibility/POSSystemStatusService.swift

```
@@ -0,0 +1,82 @@
+    /// Loads WooCommerce plugin and POS feature switch value remotely for eligibility checks.
+    /// - Parameter siteID: The site ID to fetch information for.
+    /// - Returns: POSPluginAndFeatureInfo containing plugin and feature data.
+    /// - Throws: Network or parsing errors.
+    func loadWooCommercePluginAndPOSFeatureSwitch(siteID: Int64) async throws -> POSPluginAndFeatureInfo
+}
+
+/// Contains WooCommerce plugin information and POS feature switch value.
+public struct POSPluginAndFeatureInfo {
+    public let wcPlugin: SystemPlugin?
+    public let featureValue: Bool?
+
+    public init(wcPlugin: SystemPlugin?, featureValue: Bool?) {
+        self.wcPlugin = wcPlugin
+        self.featureValue = featureValue
+    }
+}
+
+/// Service for fetching POS-related system status information.
+public final class POSSystemStatusService: POSSystemStatusServiceProtocol {
+    private let remote: SystemStatusRemote
+
+    public init(credentials: Credentials?) {
+        let network = AlamofireNetwork(credentials: credentials)
+        remote = SystemStatusRemote(network: network)
+    }
+
+    /// Test-friendly initializer that accepts a network implementation.
+    init(network: Network) {
+        remote = SystemStatusRemote(network: network)
+    }
+
+    public func loadWooCommercePluginAndPOSFeatureSwitch(siteID: Int64) async throws -> POSPluginAndFeatureInfo {
+        let mapper = SingleItemMapper<POSPluginEligibilitySystemStatus>(siteID: siteID)
+        let systemStatus: POSPluginEligibilitySystemStatus = try await remote.loadSystemStatus(
+            for: siteID,
+            fields: [.activePlugins, .settings],
+            mapper: mapper
+        )
+
+        // Finds WooCommerce plugin from active plugins response.
+        guard let wcPlugin = systemStatus.activePlugins.first(where: { $0.plugin == Constants.wcPluginPath }) else {
+            return POSPluginAndFeatureInfo(wcPlugin: nil, featureValue: nil)
```

If no WooCommerce plugin path is found at this point of execution, should we throw here rather than returning a `POSPluginAndFeatureInfo` with nil values?

---

Good question - I thought about it before, and decided to keep this service simple and just return data if available and throw errors when the app fails to retrieve the data. The use case determines how to handle when the plugin / feature value isn't available.

Ideally, all the plugin related checks can be in Yosemite. However, right now `VersionHelpers` exists in the app layer and it'd require some refactoring to move it to Yosemite. If I have some time before the i2 release next Friday, I will tackle this.